auth-source-pass.el

```diff
@@ -272,16 +272,34 @@ If ENTRIES is nil, use the result of calling `auth-source-pass-entries' instead.
                entry))
    (or entries (auth-source-pass-entries))))
 
-(defun auth-source-pass--generate-entry-suffixes (hostname user port)
+(defun auth-source-pass--generate-entry-suffixes (hostname users ports)
   "Return a list of possible entry path suffixes in the password-store.
 
 Based on the supported filename patterns for HOSTNAME, USER, &
 PORT, return a list of possible suffixes for matching entries in
 the password-store."
   (let ((domains (auth-source-pass--domains (split-string hostname "\\."))))
-    (seq-mapcat (lambda (n)
-                  (auth-source-pass--name-port-user-suffixes n user port))
-                domains)))
+       (sort-list-by-length 
+        (apply #'append 
+               (seq-mapcat (lambda (domain)
+                                   (seq-mapcat (lambda (user)
+                                                       (seq-mapcat (lambda (port)
+                                                                           (list (when (and user port)
+                                                                                   (list
+                                                                                    (format "%s@%s%s%s" user domain auth-source-pass-port-separator port)
+                                                                                    (format "%s%s%s/%s" domain auth-source-pass-port-separator port user)))
+                                                                                 (when user
+                                                                                   (list
+                                                                                    (format "%s@%s" user domain)
+                                                                                    (format "%s/%s" domain user)))
+                                                                                 (when port
+                                                                                   (list
+                                                                                    (format "%s%s%s" domain auth-source-pass-port-separator port)))
+                                                                                 (list
+                                                                                  (format "%s" domain)))
+                                                                           ) (if (consp ports) ports (list ports)))
```

Comment:
The [Emacs Lisp Style Guide](https://github.com/bbatsov/emacs-lisp-style-guide) recommends to put all closing parenthesis at the end of lines instead of the beginning.